src/Polysemy/Internal.hs

```diff
@@ -434,6 +435,18 @@ runM (Sem m) = m $ \z ->
       pure $ f $ a <$ s
 {-# INLINE runM #-}
 
+------------------------------------------------------------------------------
+-- | Type synonym for interpreters that consume an effect without changing the 
+-- return value. Offered for user convenience. 
+--
+-- 'r' Is kept polymorphic so it's possible to place constraints upon it:
+--
+-- @
+-- teletypeToIO
+-- :: 'Member' (Embed IO) r
+-- => 'InterpreterFor' Teletype r
+-- @
```

Comment:
I _think_ you need another comment line here to make the code block render properly. I don't quite get how Haddock works on that point. Also, the indentation is a bit wrong here.

To illustrate what I mean, here's how I'd do it:

```haskell
-- @r@ Is kept polymorphic so it's possible to place constraints upon it:
--
-- @
-- teletypeToIO
--   :: 'Member' (Embed IO) r
--   => 'InterpreterFor' Teletype r
-- @
--
```

Reply:
I have generated the docs and everything looked alright! I'll make this change and see how it looks!